catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -329,24 +284,21 @@ private void attemptAddRoutes() {
     LOGGER.trace(
         "Attempting to add routes for content directory monitor watching {}", monitoredDirectory);
     try {
-      RouteBuilder routeBuilder = createRouteBuilder();
+      routeBuilder = createRouteBuilder();
```

Comment:
❗️ Access to `routeBuilder` should be under protection of a lock

Reply:
@paouelle Should the init and destroy behavior be spun off into new threads so they don't block blueprint?
@paouelle If the answer to the previous question is "no", could the routeBuilder be initialized in a `static` block to prevent race conditions?

Reply:
to your first question ... yes it has to. This is what @tbatie, @peterhuffer, and I talked about this morning and Peter will be making a design change to this PR.

Reply:
I put the init and destroy method in separate threads.